src/behaviors/loading/loading.js

```diff
@@ -0,0 +1,28 @@
+(function() {
+'use strict';
+
+angular
+  .module('semantic.ui.behaviors.loading', [])
+
+  .directive('smLoading', smLoading);
+
+  smLoading.$inject = ['$parse'];
+
+  function smLoading($parse) {
+    return {
+      restrict:'A',
+      link: function(scope, element, attrs) {
+        scope.$watch(function() {
+          return $parse(attrs.smLoading)(scope);
+        }, function(loading) {
+          if (loading) {
+            element.addClass('loading');
```

Comment:
I know what it's useful for =) The point here is to try to get more value out of this directive. I think the promise API is a very good idea, specifically for the "ajax" use case you were discussing, specifically in angular. We can add it later, but it shouldn't be too difficult to add here